src/protocols/two_party_ecdsa/lindell_2017/party_one.rs

```diff
@@ -251,7 +254,14 @@ impl Signature {
         let s_tag_tag = BigInt::mod_mul(&k1_inv, &s_tag.0, &temp.q());
         let s = cmp::min(s_tag_tag.clone(), &temp.q().clone() - s_tag_tag.clone());
 
-        Signature { s, r: rx }
+        let is_ry_odd = ry.tstbit(0);
+        let mut recid = if is_ry_odd { 1 } else { 0 };
+        let half_q = &temp.q().div_floor(&BigInt::from(2));
```

Comment:
`div_floor` is not exactly q/2 (q is odd).

Reply:
modified to put this code inside `if`.